modules/unsupported/csv/src/test/java/org/geotools/data/csv/CSVWriteTest.java

```diff
@@ -0,0 +1,339 @@
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.geometry.jts.JTSFactoryFinder;
+import org.geotools.test.TestData;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.filter.Filter;
+import org.opengis.filter.FilterFactory;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.Point;
+
+/**
+ * Informal test used to document expected functionality for workshop.
+ * <p>
+ * This test has a setup method used to copy locations.csv to a temporary file.
+ * 
+ */
+public class CSVWriteTest {
+    File tmp;
+
+    File file;
+
+    @Before
+    public void createTemporaryLocations() throws IOException {
+        // Setting the system-wide default at startup time
+        System.setProperty("org.geotools.referencing.forceXY", "true");
+        
+        tmp = File.createTempFile("example", "");
+        boolean exists = tmp.exists();
+        if (exists) {
+            System.err.println("Removing tempfile " + tmp);
+            tmp.delete();
+        }
+        boolean created = tmp.mkdirs();
+        if (!created) {
+            System.err.println("Could not create " + tmp);
+            System.exit(1);
+        }
+        file = new File(tmp, "locations.csv");
+
+        URL resource = TestData.getResource(CSVWriteTest.class, "locations.csv");
+        Files.copy(resource.openStream(), file.toPath(), StandardCopyOption.REPLACE_EXISTING);
+    }
+    
+    private void fileContents( String test, File modified) throws IOException {
+        System.out.println(test +" contents start");
+        Files.copy( modified.toPath(), System.out );
+        System.out.println(test +" contents end");
+    }
+    
+    @After
+    public void removeTemporaryLocations() throws IOException {
+        File list[] = tmp.listFiles();
+        for (int i = 0; i < list.length; i++) {
+            list[i].delete();
+        }
+        tmp.delete();
+    }
+
+    @Test
+    public void featureStoreExample() throws Exception {
+        System.out.println("featureStoreExample start\n");
+        // featureStoreExample start
+        Map<String, Serializable> params = new HashMap<String, Serializable>();
+        params.put("file", file);
+        DataStore store = DataStoreFinder.getDataStore(params);
+
+        SimpleFeatureSource featureSource = store.getFeatureSource("locations");
+        if (!(featureSource instanceof SimpleFeatureStore)) {
+            throw new IllegalStateException("Modification not supported");
+        }
+        SimpleFeatureStore featureStore = (SimpleFeatureStore) featureSource;
+
+        // featureStoreExample end
+        System.out.println("\nfeatureStoreExample end\n");
+    }
+
+    @Test
+    public void transactionExample() throws Exception {
```

Comment:
Future idea: May want to migrate the transaction test from PropertyDataStore and get a real test going on here.